vtpm/Cargo.toml

```diff
@@ -0,0 +1,12 @@
+[package]
+name = "vtpm"
+edition = "2018"
+version = "0.1.0"
+
+[dependencies]
+anyhow = "1.0.56"
+byteorder = "1.4.3"
+libc = "0.2.98"
+log = "0.4.14"
+nix = "0.22.0"
```

Comment:
nix is a banned dependency. You need to reimplement whatever you need using libc crate or vmm-sys-utils.

Reply:
For my context, could you please expand on why nix is banned? Is nix banned under cloud-hypervisor project only or in Rust generally?

FYI, I removed the dependency on nix from vtpm/socket.rs module. Started working on vtpm/tpm_emulator.rs next.

Reply:
We've always tried to avoid relying on the `nix` crate given it pulls a lot of code into the project, and I think the same idea is applied in the `rust-vmm` crates. Given `nix` is a wrapper around a bunch of syscalls, we either rely on `libc` if the function is provided, or we implement what's missing ourselves.

Reply:
Thanks for the details @sboeuf. I dropped dependency on `nix` crate.